### PR TITLE
Create a gstreamer application to stream compressed video streams

### LINKED ISSUE #16: Create a camera app/frontend for Gstreamer that uses AV1 decoding

### LINKED ISSUE #202: Make av1 support More robust for camera application / test AV1 encoding
Someone with offline setup needs to test and make sure that av1 works with camera app

### PR DESCRIPTION
**Pull request by jmblixt3**
This creates 2 ros nodes a server and a client. The server node is to be run on the robot and client on drivers station laptop.
Currently rqt can be used to edit parameters, but other methods could be used in the future as well.
Please provide feedback on what features you want added

- [x] Add ability to automatically detect client ip
- [x] Add ability to change encoding/decoding method
- ~~Add bitrate/compression parameters~~
- ~~Add resolution/framerate parameters~~
- ~~Add camera src parameters~~
- ~~Test AV1 (need to be done offline since I can't do this locally)~~ pushed off into #202

### CODE REVIEW

**Comment by jmblixt3** [comment]
Fix README

**Comment by jmblixt3** [comment]
@Isopod00 test on private network

**Comment by jmblixt3** [comment]
Ready to merge

File: README.md [hunk 1]
```diff
@@ -194,45 +194,23 @@ sudo grep 'mycmd' /var/log/syslog
...
+To start gstreamer client make sure to add the deepstream layer to the docker layers 
 
-```
-gst-launch-1.0 v4l2src device=/dev/video0 ! "video/x-raw,width=640,height=480,framerate=15/1" ! nvvidconv ! "video/x-raw(memory:NVMM),format=NV12" ! nvv4l2av1enc bitrate=200000 ! "video/x-av1" ! udpsink host=127.0.0.1 port=5000
-```
-
-Start Gstreamer AV1 Decoding (On Nvidia Jetson AGX Orin): 
-
-```
-gst-launch-1.0 udpsrc port=5000 ! "video/x-av1,width=640,height=480,framerate=15/1" ! queue ! nvv4l2decoder ! nv3dsink
```
Comment (Isopod00) [inline README.md:206]: Why did you delete all these commands from the README @jmblixt3? Because I feel like they could be useful to keep around in case they are needed some time.

Comment (jmblixt3) [reply README.md:206]: Go look at an old revision they are now obsolete

Comment (Isopod00) [reply README.md:206]: please keep them as a reference, you can move them to a different drop-down menu of the readme if you want, called "gstreamer pipeline examples" or something

**Review by pwmarshall — APPROVED** [review]
Tested with jetson and works :)
